core/lib/zksync_core/src/eth_watch/event_processors/set_chain_id.rs

```diff
@@ -0,0 +1,73 @@
+use zksync_contracts::SET_CHAIN_ID_EVENT;
+use zksync_dal::StorageProcessor;
+use zksync_types::{protocol_version::decode_set_chain_id_event, web3::types::Log, Address, H256};
+
+use crate::eth_watch::{
+    client::{Error, EthClient},
+    event_processors::EventProcessor,
+    metrics::{PollStage, METRICS},
+};
+
+/// Responsible for saving `setChainId` upgrade transactions to the database.
+#[derive(Debug)]
+pub struct SetChainIDEventProcessor {
+    /// Address of the `DiamondProxy` contract of the chain that the `SetChainId` event is for.
+    diamond_proxy_address: Address,
+    /// Signature of the `SetChainIdUpgrade` event.
+    /// The event is emitted by the `StateTransitionManager` contract.
+    set_chain_id_signature: H256,
+}
+
+impl SetChainIDEventProcessor {
```

Comment:
Bikeshedding: Conventionally, acronyms in Rust are spelled with just the first letter capitalized (i.e., `SetChainIdEventProcessor`).